File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        x=0
        temp=0
        while(n>0):
            y=n%10
            x=x+(pow(y,2))
            n=n//10
            if n==0:
                n=x
                temp=x
                x=0
            if temp==1:
                return True
            if temp==4:
                return False
            
        

File: 0202-happy-number/0202-happy-number.java
class Solution {
    static int func(int a){
        int r=0;
        while(a>0){
            int d=a%10;
            a=a/10;
            r+=d*d;
        }
        return r;
    }
    public boolean isHappy(int n) {
        int s=n;
        int f=n;
        while(f!=1){
            s=func(s);
            f=func(f);
            f=func(f);
            if(s==f){
                while(s==f && s!=1) return false;
            }
        }return true;

        
    }
}